src/java_bytecode/java_bytecode_instrument.cpp

```diff
@@ -132,6 +136,32 @@ codet java_bytecode_instrumentt::throw_exception(
   return init_code;
 }
 
+
+/// Checks whether there is a division by zero
+/// and throws ArithmeticException if necessary.
+/// Exceptions are thrown when the `throw_runtime_exceptions`
+/// flag is set.
+/// \return Based on the value of the flag `throw_runtime_exceptions`,
+/// it returns code that either throws an ArithmeticException
+/// or is a skip
+codet java_bytecode_instrumentt::check_arithmetic_exception(
+  const exprt &denominator,
+  const source_locationt &original_loc)
+{
+  symbolt exc_symbol;
+
```

Comment:
as only integral div-by-zero throws, maybe add invariant for `java_int_type` or `java_long_type`